streaming/src/ring_buffer.h

```diff
@@ -78,7 +78,6 @@ class StreamingTransientBuffer {
 template <class T>
 class AbstractRingBufferImpl {
```

Comment:
Not quite related to this PR, but it would be nice to rename this to AbstractRingBuffer as there's no implementation.